lib/options.nix

```diff
@@ -6,6 +6,34 @@
 with lib;
 with nixvimUtils;
 rec {
+  # Render a plugin default string
+  pluginDefaultText =
+    {
+      # plugin default, used if `defaultText` is missing
+      default ? null,
+      # pre-formatted plugin default string
+      #
+      # Assume a string `default` is already formatted as intended,
+      # TODO: remove this behavior so we can quote strings properly
+      # historically strings were the only type accepted by mkDesc.
+      defaultText ? if isString default then default else generators.toPretty { } default,
+      # nix option default value, used if `nixDefaultText` is missing
+      nixDefault ? null,
+      # pre-formatted nix option default string
+      nixDefaultText ? generators.toPretty { } nixDefault,
+      ...
+    }:
+    let
+      # Format text using markdown code block or inline code
+      # TODO support `s` being a literalExpression or literalMD type
+      toMD = s: if hasInfix "\n" s then "\n\n```nix\n${s}\n```" else "`${s}`";
+    in
+    literalMD ''
+      ${toMD nixDefaultText}
```

Comment:
Upstream's docs rendering would actually pass `defaultText` through `toPretty` and then wrap it in `literalExpression`. (Unless `defaultText` is already a `literalExpression` or `literalMD` type).

This means passing `defaultText` will usually render as a quoted string, unlike my implementation.

It might be better to more closely replicate the upstream implementation, so that the behavior isn't surprising. I guess set the fallback value to something like:
```nix
# legacy isString special-case omitted for clarity
defaultText ? literalExpression (generators.toPretty { } default),
nixDefaultText ? literalExpression (generators.toPretty { } nixDefault),
```

And then correctly handle literal expressions in the main implementation.

Reply:
I also noticed upstream pass `allowPrettyValues = true` to `toPretty`, this enables support for embedded custom rendering. Any value of the type `{ __pretty = fn; val = …; }` will use the `__pretty` function to pretty print `val`.

e.g. you could nest `{ __pretty = v: v; val = "(raw nix expression)"; }` somewhere in a list/set, and it'd be pretty-printed as `(raw nix expression)`!

Reply:
This is resolved, but for now I'll leave the thread visible for context.